internal/service/alarms/internal/alertmanager/alertmanager.go

```diff
@@ -0,0 +1,160 @@
+
+	// TODO: Alerts flipping between Resolved and Firing status in a HA setup.
+	// https://issues.redhat.com/browse/ACM-13481
+	// Replicas of the alertmanager statefulset can be modified from the spec of the observability object (MultiClusterObservability Kind)
+	// spec.advanced.alertmanager.replicas
+	resolveTimeout = "5m"
+	groupWait      = "30s"
+	groupInterval  = "1m"
+	repeatInterval = "4h"
+
+	// Main receiver and null receiver to filter out alerts
+	oranAlarmReceiver = "oran_alarm_receiver"
+	nullReceiver      = "null"
+)
+
+var (
+	// Labels to identify a cluster
+	clusterIDLabels = []string{"managed_cluster", "clusterID"}
+	// Alerts to filter out
+	skipAlerts = []string{"Watchdog"}
+)
+
+type Config struct {
+	ResolveTimeout string
+	Route          RouteConfig
+	Receivers      []ReceiverConfig
+}
+
+type RouteConfig struct {
+	GroupWait      string
+	GroupInterval  string
+	RepeatInterval string
+	Receiver       string
+	GroupBy        []string
+	Matchers       []string
+
+	Routes []RouteConfig
+}
+
+type ReceiverConfig struct {
+	Name         string
+	SendResolved bool
+	URL          string
+}
+
+// Setup updates the alertmanager config secret with the new configuration
+func Setup(ctx context.Context, cl client.Client, port string) error {
+	// ACM recreates the secret when it is deleted, so we can safely assume it exists
+	var secret corev1.Secret
+	err := cl.Get(ctx, client.ObjectKey{Namespace: namespace, Name: secretName}, &secret)
+	if err != nil {
+		return fmt.Errorf("failed to get secret %s/%s: %w", namespace, secretName, err)
+	}
+
+	// Verify that Data has the key "alertmanager.yaml"
+	if _, ok := secret.Data[secretKey]; !ok {
+		return fmt.Errorf("%s not found in secret %s/%s", secretKey, namespace, secretName)
+	}
+
+	configFile, err := generateConfigFile(port)
+	if err != nil {
+		return fmt.Errorf("failed to generate alertmanager config file: %w", err)
+	}
+
+	secret.Data[secretKey] = configFile.Bytes()
+	err = cl.Update(ctx, &secret)
+	if err != nil {
+		return fmt.Errorf("failed to update secret %s/%s: %w", namespace, secretName, err)
+	}
+
+	return nil
+}
+
+// generateConfigFile generates the alertmanager configuration file
+func generateConfigFile(port string) (*bytes.Buffer, error) {
+	// Receivers to send alerts to
+	receivers := []ReceiverConfig{
+		{Name: nullReceiver},
+		{
+			Name:         oranAlarmReceiver,
+			SendResolved: true,
+			URL:          fmt.Sprintf(URL, port),
+		},
+	}
+
+	routes := []RouteConfig{}
+	// Route to filter out alerts
+	r := RouteConfig{Receiver: nullReceiver}
+	for _, alert := range skipAlerts {
+		r.Matchers = append(r.Matchers, fmt.Sprintf("alertname=%s", alert))
+	}
+	routes = append(routes, r)
+
+	// Routes to group alerts by cluster identifier
+	for _, label := range clusterIDLabels {
+		r = RouteConfig{Receiver: oranAlarmReceiver}
+		r.GroupBy = append(r.GroupBy, label)
+		r.Matchers = append(r.Matchers, fmt.Sprintf("%s=~\".+\"", label))
+		routes = append(routes, r)
+	}
```

Comment:
Is there chance we may miss alerts if we start adding filter at this stage? I think we should accept all the alerts now and then start adding filter in the subsequent PRs?

Reply:
These will group by labels that identify the cluster. If these labels do not exist, the alert is sent anyway due to the parent receiver.  So, we won't miss anything.

Reply:
Gotcha! All good as long as we dont miss anything. 

But just a heads up - i think we are guaranteed to get `managed_cluster_id` and the alert name for all the notifications. If that's not the case we might need to reach out to acm folks right? @browsell

Reply:
I've seen alerts without the label `managed_cluster`. They had instead the label `clusterID`. That's why I'm creating a group for it. I will report it to see if it is a bug.

Reply:
Are those from the local cluster ?

Reply:
don't remember exactly but alerts coming from the local cluster also come with the `managed_cluster` label